subsys/bluetooth/host/att.c

```diff
@@ -3756,6 +3759,14 @@ void bt_att_set_tx_meta_data(struct net_buf *buf, bt_gatt_complete_func_t func,
 
 	data->func = func;
 	data->user_data = user_data;
+	data->attr_count = 1;
+}
+
+void bt_att_increment_tx_meta_data_attr_count(struct net_buf *buf, uint16_t attr_count)
```

Comment:
I think we should just inline the code in this function. It leads to some not-beautiful code; like `num_params-1`. Casting user data is a somewhat unsafe operation, that can benefit from visibility. It can be expressed as a simple one-liner `bt_att_tx_meta_data(buf)->attr_count +=`. And the reasoning employed to justify the lack of overflow protection implies that this function shall only be used for notify multi, so it is of limited use.